docs/rpc/openapi.yaml

```diff
@@ -551,3 +551,41 @@ paths:
             type: string
           description: The Stacks chain tip to query from. If tip == latest, the query will be run from the latest
             known tip (includes unconfirmed state).
+
+  /v2/block_proposal:
+    post:
+      summary: Validate a proposed Stacks block
+      tags:
+        - Mining
+      #operationId: ???
+      description: |
+        Used by miner to validate a proposed Stacks block.
+        Can accept either JSON or binary encoding.
+        
+        **This endpoint will only accept requests over the local loopback network interface.**
+      responses:
+        202:
+          description: Block proposal is valid
+        400:
+          description: Bad request
```

Comment:
Can you enumerate the reasons why this error will be reached?

Reply:
This specific error just means that what was submitted wasn't a valid block proposal (would not be accepted be a Stacks node). I changed it to make that more clear, but there are too many things that could make a block invalid to enumerate them individually